conda_forge_tick/status_report.py

```diff
@@ -118,7 +118,8 @@ def write_version_migrator_status(migrator, mctx):
                         )
 
     with open("./status/version_status.json", "w") as f:
-        old_out = out.copy()
+        old_out: dict[str, dict[str, str] | set[str]] = {}
+        old_out.update(out)
```

Comment:
```
conda_forge_tick/status_report.py:121: error: Incompatible types in assignment (expression has type "dict[str, dict[str, str]]", variable has type "dict[str, dict[str, str] | set[str]]")  [assignment]
conda_forge_tick/status_report.py:121: note: "Dict" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
conda_forge_tick/status_report.py:121: note: Consider using "Mapping" instead, which is covariant in the value type
```

Changing the type of `out` is wrong because it then causes trouble above.

Reply:
(`out` cannot have sets as values)